core/src/main/java/hudson/tasks/BuildTrigger.java

```diff
@@ -182,6 +194,61 @@ public boolean hasSame(Collection<? extends AbstractProject> projects) {
 
     @Override
     public boolean perform(AbstractBuild build, Launcher launcher, BuildListener listener) {
+        List<Job<?, ?>> jobs = new ArrayList<>();
+        for (Job<?, ?> job : getChildJobs(build.getProject())) {
+            if (job instanceof AbstractProject) {
+                continue; // taken care of by DependencyGraph
+            }
+            jobs.add(job);
+        }
+        if (!jobs.isEmpty() && build.getResult().isBetterOrEqualTo(threshold)) {
+            PrintStream logger = listener.getLogger();
+            Authentication auth = Jenkins.getAuthentication();
+            if (auth.equals(ACL.SYSTEM)) {
```

Comment:
This block could perhaps be dropped as part of a more general cleanup discussed in [JENKINS-22949](https://issues.jenkins-ci.org/browse/JENKINS-22949), which I will investigate separately.